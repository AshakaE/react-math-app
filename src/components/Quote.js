import React, { useEffect, useState } from 'react';
import Navbar from './Navbar';
import getQuote from './getQuote';

const Quote = () => {
  const [quote, setQuote] = useState('');

  useEffect(() => {
    getQuote().then((items) => {
      const random = Math.floor(Math.random() * items.length);
      const qday = items[random];
      if (quote === '') {
        setQuote(qday);
      }
    });
  }, []);

  return (
    <>
      <div>
        <Navbar />
      </div>
      <div className="m-auto px-56 mt-60 text-center">
        <div>
          <p className="text-4xl font-extralight italic">{quote.text}</p>
          <h4 className="text-3xl pt-3">{quote.author ?? 'Anonymous'}</h4>
        </div>
      </div>
    </>
  );
};

export default Quote;
