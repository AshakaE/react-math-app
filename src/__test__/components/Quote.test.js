import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import Quote from '../../components/Quote';

describe('Quote', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <Quote />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
