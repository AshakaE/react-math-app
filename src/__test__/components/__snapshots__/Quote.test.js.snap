// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote renders correctly 1`] = `
Array [
  <div>
    <div
      className="flex justify-between px-28 mt-10 mb-16"
    >
      <div>
        <h1
          className="text-4xl font-medium uppercase"
        >
          Math Magicians
        </h1>
      </div>
      <div
        className="text-2xl flex w-96 justify-between underline text-blue-700"
      >
        <a
          className="hover:text-blue-400 duration-150"
          href="/"
          onClick={[Function]}
        >
          Home 
        </a>
        <a
          className="hover:text-blue-400 duration-150"
          href="/calculator"
          onClick={[Function]}
        >
          Calculator 
        </a>
        <a
          className="hover:text-blue-400 duration-150"
          href="/quote"
          onClick={[Function]}
        >
          Quote 
        </a>
      </div>
    </div>
  </div>,
  <div
    className="m-auto px-56 mt-60 text-center"
  >
    <div>
      <p
        className="text-4xl font-extralight italic"
      />
      <h4
        className="text-3xl pt-3"
      >
        Anonymous
      </h4>
    </div>
  </div>,
]
`;
